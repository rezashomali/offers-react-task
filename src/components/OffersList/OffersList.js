import { Grid } from "@material-ui/core";
import OffersListItem from "../OffersListItem/OffersListItem";
import { useSelector } from "react-redux";

const OffersList = () => {
  const offers = useSelector((state) => state.offers);

  return (
    <Grid container spacing={2} justify="center">
      {offers.map((offer) => (
        <OffersListItem key={offer.id} offer={offer} />
      ))}
    </Grid>
  );
};

export default OffersList;
