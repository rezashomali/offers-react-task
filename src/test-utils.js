import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { offersReducer } from "./redux/reducers/offersReducer";

function render(
  ui,
  {
    initialState,
    store = createStore(offersReducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";
// override render method
export { render };
