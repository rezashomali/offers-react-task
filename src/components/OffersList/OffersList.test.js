import { render } from "../../test-utils";
import OffersList from "./OffersList";

it("OffersList snapshot", () => {
  const initialState = {
    loading: true,
    offers: [],
    error: "",
  };

  const container = render(<OffersList />, { initialState: initialState });
  expect(container.firstChild).toMatchSnapshot();
});
