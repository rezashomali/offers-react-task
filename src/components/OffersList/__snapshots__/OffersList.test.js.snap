// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OffersList snapshot 1`] = `undefined`;
